src/NHibernate/Util/LinkHashMap.cs

```diff
@@ -0,0 +1,650 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.Serialization;
+using System.Text;
+using NHibernate.DebugHelpers;
+
+namespace NHibernate.Util
+{
+	/// <summary>
+	/// A map of objects whose mapping entries are sequenced based on the order in which they were
+	/// added. This data structure has fast <c>O(1)</c> search time, deletion time, and insertion time
+	/// </summary>
+	/// <remarks>
+	/// This class is not thread safe.
+	/// This class is not really a replication of JDK LinkedHashMap{K, V}, 
+	/// this class is an adaptation of SequencedHashMap with generics.
+	/// </remarks>
+	[DebuggerTypeProxy(typeof(CollectionProxy<>))]
+	[Serializable]
+	internal class LinkHashMap<TKey, TValue> : IDictionary<TKey, TValue>, IDeserializationCallback
+	{
+		[Serializable]
+		protected class Entry
+		{
+			private readonly TKey key;
+			private TValue evalue;
+			private Entry next;
+			private Entry prev;
```

Comment:
To avoid the `this`, switch to `_` prefix for class members.

```suggestion
			private readonly TKey _key;
			private TValue _value;
			private Entry _next;
			private Entry _prev;
```

Reply:
I thought it was o code style :)))

Reply:
applied

Reply:
There are no [Microsoft guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-fields) for private fields. `camelCase` naming without `_` prefix in NHibernate is generally the result of port from Java without renaming. This causes the fields to be looking like local variables and methods parameters. The `_` prefix is deemed ugly by some, but has the merit to allow obvious recognition of fields versus local variables and method parameters.